thorlcr/graph/thgraph.cpp

```diff
@@ -2674,12 +2674,9 @@ void CJobBase::startJob()
 {
     LOG(MCdebugProgress, thorJob, "New Graph started : %s", graphName.get());
     ClearTempDirs();
-    unsigned pinterval = globals->getPropInt("@system_monitor_interval",1000*60);
-    if (pinterval)
-    {
-        perfmonhook.setown(createThorMemStatsPerfMonHook(*this, getOptInt(THOROPT_MAX_KERNLOG, 3)));
-        startPerformanceMonitor(pinterval,PerfMonStandard,perfmonhook);
-    }
+    setPerformanceMonitorHook(nullptr);
```

Comment:
> doubt startJob() would ever be called again before endJob()

No, it won't. Would be a bug if so. It would imply the job (CJobBase had leaked).

The 'setPerformanceMonitorHook(nullptr);' here implies to someone reading the code that it's normal / to be expected, which it is not.

>  didn't want a perf thread using the hook while in the next line it could be changed. 

chaging perfmonhook, wouldn't invalidate the old installed hook. The perf. thread would only use the new hook after setHook had left it's CS.
The perf thread _could_ have been calling the old hook at any point prior to this too, but only if/because something else had gone wrong and the job had leaked.

I think you're fine with just the lines below. Or you get setPerformanceMonitorHook to return any pre-existing hook and assert that null.